sumpy/p2p.py

```diff
@@ -480,30 +487,51 @@ def get_kernel(self):
                 <> isrc_box_start = source_box_starts[itgt_box]
                 <> isrc_box_end = source_box_starts[itgt_box+1]
 
-                for isrc_box
+                for ipoint
+                  <> itgt = ipoint + itgt_start
+                  <> cond_itgt = itgt < itgt_end
+                  <> acc[iknl] = 0 {id=init_acc, dup=iknl}
+                  if cond_itgt
+                    tgt_center[idim] = targets[idim, itgt] {id=load_tgt, dup=idim}
+                  end
+                  for isrc_box
                     <> src_ibox = source_box_lists[isrc_box]
                     <> isrc_start = box_source_starts[src_ibox]
                     <> isrc_end = isrc_start + box_source_counts_nonchild[src_ibox]
-
-                    for itgt
-                    for isrc
-                        <> d[idim] = \
-                            targets[idim, itgt] - sources[idim, isrc]
+                    <> cond_isrc = ipoint < isrc_end - isrc_start
+                    if cond_isrc
+                      local_isrc[idim, ipoint] = sources[idim,
+                          ipoint + isrc_start]  {id=load_src, dup=idim}
+                      local_isrc_strength[istrength, ipoint] = strength[istrength,
+                          ipoint + isrc_start]  {id=load_charge}
```

Comment:
Can't `add_prefetch` add this?

Reply:
I tried doing that, but I'm not sure I was doing it correct.
I get a,
```
RuntimeError: Inames 'idim, isrc' require more than one leaf domain, which makes the domain change that is part of your current operation ambiguous.
```

with `knl = lp.add_prefetch(knl, "sources")`.
Any ideas?

Reply:
If you duplicate the `idim` to the point where it's only used by this statement, this should resolve itself.

Reply:
```
>               raise RuntimeError("no invocations of '%s' found" % subst_name)
E               RuntimeError: no invocations of 'sources_fetch_rule' found

../loopy/loopy/transform/precompute.py:550: RuntimeError
```